### ISSUE TITLE
Fix bug flake8 --diff - [merged]

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#919]
In GitLab by @glemaitre on Jan 30, 2019, 08:48

_Merges is/490 -> master_

closes #490 

`style_guides` is a list and should not call the method `values` but just iterate over the style guides instead.

**Comment by asottile** [comment]
In GitLab by @codecov on Jan 30, 2019, 08:49

# [Codecov](https://codecov.io/gl/pycqa/flake8/pull/292?src=pr&el=h1) Report
> Merging [#292](https://codecov.io/gl/pycqa/flake8/pull/292?src=pr&el=desc) into [master](https://codecov.io/gl/pycqa/flake8/commit/f70ae10c2ba191bef78c723a3404c27d84f13de8?src=pr&el=desc) will **increase** coverage by `6.67%`.
> The diff coverage is `100%`.

[![Impacted file tree graph](https://codecov.io/gl/pycqa/flake8/pull/292/graphs/tree.svg?width=650&token=QL6TVnaXih&height=150&src=pr)](https://codecov.io/gl/pycqa/flake8/pull/292?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #292      +/-   ##
==========================================
+ Coverage    71.9%   78.58%   +6.67%     
==========================================
  Files          25       25              
  Lines        2367     2367              
  Branches      393      393              
==========================================
+ Hits         1702     1860     +158     
+ Misses        584      421     -163     
- Partials       81       86       +5
```


| [Impacted Files](https://codecov.io/gl/pycqa/flake8/pull/292?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/flake8/style\_guide.py](https://codecov.io/gl/pycqa/flake8/pull/292/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC9zdHlsZV9ndWlkZS5weQ==) | `98% <100%> (+4.5%)` | :arrow_up: |
| [src/flake8/utils.py](https://codecov.io/gl/pycqa/flake8/pull/292/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC91dGlscy5weQ==) | `82.63% <0%> (+1.05%)` | :arrow_up: |
| [src/flake8/options/config.py](https://codecov.io/gl/pycqa/flake8/pull/292/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC9vcHRpb25zL2NvbmZpZy5weQ==) | `96.51% <0%> (+2.32%)` | :arrow_up: |
| [src/flake8/formatting/base.py](https://codecov.io/gl/pycqa/flake8/pull/292/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC9mb3JtYXR0aW5nL2Jhc2UucHk=) | `75.43% <0%> (+5.26%)` | :arrow_up: |
| [src/flake8/plugins/pyflakes.py](https://codecov.io/gl/pycqa/flake8/pull/292/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC9wbHVnaW5zL3B5Zmxha2VzLnB5) | `62.68% <0%> (+10.44%)` | :arrow_up: |
| [src/flake8/main/application.py](https://codecov.io/gl/pycqa/flake8/pull/292/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC9tYWluL2FwcGxpY2F0aW9uLnB5) | `73.68% <0%> (+14.21%)` | :arrow_up: |
| [src/flake8/checker.py](https://codecov.io/gl/pycqa/flake8/pull/292/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC9jaGVja2VyLnB5) | `73.16% <0%> (+16.93%)` | :arrow_up: |
| [src/flake8/processor.py](https://codecov.io/gl/pycqa/flake8/pull/292/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC9wcm9jZXNzb3IucHk=) | `89.12% <0%> (+22.17%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gl/pycqa/flake8/pull/292?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gl/pycqa/flake8/pull/292?src=pr&el=footer). Last update [f70ae10...915184a](https://codecov.io/gl/pycqa/flake8/pull/292?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by asottile** [comment]
In GitLab by @glemaitre on Jan 30, 2019, 08:49

Let's work the tests out :)

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 09:48

I wrote an integration test, though I can't push to your branch :( (there's a checkbox in the PR iirc):

```diff
commit 4b0d5e2330008f1db144c0e5c1e20af3f7e7ecbb (HEAD -> is/490)
Author: Anthony Sottile <asottile@umich.edu>
Date:   Wed Jan 30 09:47:48 2019 -0800

    Add integration test for `flake8 --diff`

diff --git a/tests/integration/test_application.py b/tests/integration/test_application.py
new file mode 100644
index 0000000..fe528d2
--- /dev/null
+++ b/tests/integration/test_application.py
@@ -0,0 +1,43 @@
+import mock
+
+from flake8.main import application
+from flake8 import utils
+
+
+def test_diff_option(tmpdir, capsys):
+    """Ensure that FileChecker can handle --diff"""
+
+    t_py_contents = '''\
+import os
+import sys  # unused but not part of diff
+
+print('(to avoid trailing whitespace in test)')
+print('(to avoid trailing whitespace in test)')
+print(os.path.join('foo', 'bar'))
+
+y  # part of the diff and an error
+'''
+
+    diff = '''\
+diff --git a/t.py b/t.py
+index d64ac39..7d943de 100644
+--- a/t.py
++++ b/t.py
+@@ -4,3 +4,5 @@ import sys  # unused but not part of diff
+ print('(to avoid trailing whitespace in test)')
+ print('(to avoid trailing whitespace in test)')
+ print(os.path.join('foo', 'bar'))
++
++y  # part of the diff and an error
+'''
+
+    with mock.patch.object(utils, 'stdin_get_value', return_value=diff):
+        with tmpdir.as_cwd():
+            tmpdir.join('t.py').write(t_py_contents)
+
+            app = application.Application()
+            app.run(['--diff'])
+
+    out, err = capsys.readouterr()
+    assert out == "t.py:8:1: F821 undefined name 'y'\n"
+    assert err == ''
```

**Comment by asottile** [comment]
In GitLab by @glemaitre on Jan 30, 2019, 09:49

changed the description

**Comment by asottile** [comment]
In GitLab by @glemaitre on Jan 30, 2019, 09:50

I think that I checked the box now.

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 09:54

added 1 commit

<ul><li>915184a1 - Add integration test for `flake8 --diff`</li></ul>

[Compare with previous version](https://gitlab.com/pycqa/flake8/merge_requests/292/diffs?diff_id=32790684&start_sha=763f68b6233380105cc2ae6565ce955ba7de6b8e)

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 09:59

unmarked as a **Work In Progress**

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 10:01

approved this merge request

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 10:09

mentioned in commit 941b1208193ec26b7958aa8bf3ad4221fc813e91

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 10:09

merged

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 10:10

Thanks for the fix!  I'll get this released as 3.7.2 soon
